README.md

```diff
@@ -0,0 +1,56 @@
+XMR to BTC Atomic Swap
+======================
+
+This repository is a proof of concept for atomically swapping XMR for BTC.
+
+We define:
+
+- Alice to be the actor that initially holds XMR.
+- Bob to be the actor that initially holds BTC.
+
+The repository is structured as a library and a single test function that executes the swap.
+The library has the following modules:
+
+- `alice`: Defines the state machine that describes the swap for Alice.
+This includes the messages sent to/from Alice.
+- `bob`: Defines the state machine that describes the swap for Bob.
+This includes the messages sent to/from Bob.
+- `bitcoin`: Keys, signing functions, transactions etc. for Bitcoin.
+Also includes a test wallet (see below).
+- `monero`: Keys, signing functions, transactions etc. for Monero.
+Also includes a test wallet (see below).
+
+Currently we have a single test function that proves the following:
+
+- Interaction with both block chains and their respective wallets works.
+- The messages required are correct and can manually drive the state transitions to execute a swap.
+  
+
+Currently we do not do:
+
+- Actual network communication.
+- Watch the blockchain for transactions (we just assume they have been mined as soon as we broadcast and move onto the next state).
```

Comment:
I think we do that in a simplified way, i.e. poll once and fail if the transaction is not there.

Reply:
Polling until failure (watching) has been implemented in the latest PR. However this PR is targetting master.

Reply:
Oh yeah, I'm totally wrong here. Thanks, will fix.

FTR @rishflab this PR targets master. If it goes in before yours then the obligation is on you to update the README as part of your PR. If yours goes in first then the obligation is on me to update this PR. Thems the breaks.